meta-lmp-base/recipes-core/initrdscripts/initramfs-module-install-efi/init-install-efi.sh

```diff
@@ -9,6 +9,20 @@
 
 PATH=/sbin:/bin:/usr/sbin:/usr/bin
 
+check_secure_boot() {
+    if [ ! -d /sys/firmware/efi/efivars ]; then
+        echo "EFI vars sysfs mount point not found"
+	exit 1
+    fi
+
+    efi_secure=`hexdump /sys/firmware/efi/efivars/SecureBoot-* | head -n1 | awk '{print $4}'`
```

Comment:
the efivar tool should be more bullet proof instead of parsing binary strings.
It has no dependencies, so its installation size is super small.

```
# Whether the platform firmware is operating in Secure boot mode (1) or not (0)
efi_secure=`efivar --name=8be4df61-93ca-11d2-aa0d-00e098032b8c-SecureBoot --print-decimal`
# Whether the system should require authentication on SetVariable() requests to Secure Boot policy variables (0) or not (1)
efi_mode=`efivar --name=8be4df61-93ca-11d2-aa0d-00e098032b8c-SetupMode --print-decimal`
```

The `8be4df61-93ca-11d2-aa0d-00e098032b8c` is a global variable and part of the spec
```
efivar --list-guids|grep 8be4df61-93ca-11d2-aa0d-00e098032b8c
{8be4df61-93ca-11d2-aa0d-00e098032b8c}  {global}        efi_guid_global EFI Global Variable
```